src/hivpy/population.py

```diff
@@ -66,6 +75,62 @@ def _create_population_data(self):
             self.data[col.HIV_STATUS] = self.hiv_status.introduce_HIV(self.data)
             self.HIV_introduced = True
 
+    def init_variable(self, name: str, init_val, n_prev_steps=0):
+        """
+           New variable will be initialised as a collection of columns.\\
+           Column names (keys) will be (name, 0) ... (name, n_prev_steps).\\
+           Updates dictionary to keep track of number of time steps being stored for each
+           variable.
+           name: string, name of variable
+           n_prev_steps: integer, number of previous iterations of this variable which need
+           to be stored (default 0).
+        """
+        self._variable_history[name] = n_prev_steps
+        for i in range(0, n_prev_steps + 1):
+            self.data[(col.name, i)] = init_val + 1
+
+    def get_sub_pop(self, conditions):
+        index = reduce(operator.and_,
+                       (op(self.data[self.get_correct_column(var)], val) for (var, op, val) in conditions))
+        return index
+
+    def apply_vector_func(self, params, func):
+        param_cols = list(map(self.get_correct_column, params))
+        return func(*param_cols)
+
+    def get_variable(self, var, sub_pop=None):
+        var_col = self.get_correct_column(var)
+        if sub_pop is None:
+            return self.data[var_col]
+        else:
+            return self.data.loc[sub_pop, var_col]
+
+    def set_present_variable(self, target: str, value, sub_pop=None):
+        present_col = self.get_correct_column((target, 0))
+        if sub_pop is None:
+            self.data[present_col] = value
+        else:
+            self.data.loc[sub_pop, present_col] = value
+
+    def get_correct_column(self, param_info):
+        (param, dt) = self.make_column_tuple(param_info)
+        col_index = (self.step + dt) % self.variable_history[param]
+        return (param, col_index)
+
+    def set_variable_by_group(self, target, groups, func, use_size=True, sub_pop=None):
+        """Sets the value of a population variable at the present time step by calling transform group."""
+        target_col = self.get_correct_column((target, 0))
+        if sub_pop is None:
+            self.data[target_col] = self.transform_group(groups, func, use_size)
+        else:
+            self.data.loc[sub_pop, target_col] = self.transform_group(groups, func, use_size, sub_pop)
+
+    def make_column_tuple(param):
+        if (type(param) == str):
+            return (param, 1)
```

Comment:
Should the default value be 0 or 1 (as in the setting methods)?

Reply:
I chose the default of 1 because usually we are getting columns from t-1 in order to calculate new values, but I admit it's not super clear so maybe we shouldn't have a default at all.